### ISSUE TITLE
Party leader promotion whitelist

### ISSUE THREAD

**Issue by Dedoooo** [NadyaNayme/TidyChat#48]

**Comment by NadyaNayme** [comment]
Just some notes for myself.

Pretty sure this is the one: https://xivapi.com/LogMessage/23?pretty=true

No idea what this one is for but it could also be this one I guess: https://xivapi.com/LogMessage/15?pretty=true

Soonest this will be added is in the next release in 2 weeks. Everyone chose until _after_ I did the release today to start making new filter requests. >-<

**Comment by Dedoooo** [comment]
Thank you

**Comment by Dedoooo** [comment]
if there's anything i can do to help out with figuring which filter is which i'm up for it. can build it locally if you merge or you can dm me on discord

**Comment by Dedoooo** [comment]
also could you possibly add a bonus whitelist for duties? apparently its filtered out
basically the info when you join a duty with someone new to it with bonus

**Comment by NadyaNayme** [comment]
If you'd like to help the best way is to provide me a screenshot of the exact message you want a filter for and provide me with the message so I can find it on Xivapi easier. I don't really mind adding them but if I don't know what I'm searching for finding them in XivApi is a bit of a pain and I need to do that so I can make sure that the filters are localized for JP/DE/FR clients.

X/Y party members are eligible for duty rewards. The number of coffers to appear will be...
https://xivapi.com/LogMessage/4233?pretty=true
https://xivapi.com/LogMessage/4238?pretty=true
https://xivapi.com/LogMessage/4246?pretty=true

All of the above are very similar so I'll just write my regex to filter all 3 of them since I don't know in which contexts one is used over the other.
 
Bonus reward for completing someone's first clear: https://xivapi.com/LogMessage/4225?pretty=true

And I guess I should probably add earning Second Chance points: https://xivapi.com/LogMessage/7975?pretty=true

**Comment by NadyaNayme** [comment]
So this one is a little wonky because the Bonus Awarded for adventurer in need and using duty roulette are Progress type messages IIRC (which is why they're in the Progress tab) while First Clear awards are System-type messages at the start of a duty. I don't want to split these up because they're all sorta related? Maybe I should move these to System under "Bonus Award Messages"? I don't really like that either. It's a choice between being confusing or being wrong (which can be confusing for the more engaged users who know their message types based on chat styling...)

![image](https://user-images.githubusercontent.com/4567166/169188587-eee0f407-391b-44a9-b9f3-61cdcd7eb6e2.png)


![image](https://user-images.githubusercontent.com/4567166/169188576-d5973e29-d10c-4619-bdd7-7724360cbf2b.png)

![image](https://user-images.githubusercontent.com/4567166/169188604-53feae88-1296-4338-8693-60653897509d.png)

Also debating on getting rid of all of the redundant "messages" terminology but it helps keep things a bit consistently worded.

Again - these filters won't actually be available until Tidy Chat's next update on the 31st.

**Comment by Dedoooo** [comment]
Hello, i've made a local build to test the changes with said settings allowed through the filter and they still do not show in the chat for me, however, when i disabled the main "Filter system spam" they appear. could they have a conflict maybe? 

![image](https://user-images.githubusercontent.com/93833740/169205354-ae08bf3e-004d-457f-9ecf-9fc0444ae7b7.png)
![image](https://user-images.githubusercontent.com/93833740/169205403-1541e740-0a7e-4307-8e1b-e2740738c7b7.png)

i've not tried going into savage yet but i'll try that later.

sorry if i'm bothering you with this too much 😅 i appreciate your work! thank you.

**Comment by NadyaNayme** [comment]
I appreciate it. There's 2 weeks to iron out the bugs in the filters and the problematic one is definitely the first party ones - as I'm trying to avoid conflicts with other "One or more party members have yet to complete this duty." messages.

The first screenshot is a toast message which Tidy Chat does not block - you would use Burnt Toast to filter it. The actual chat message is problematic because I forgot `.` doesn't match line breaks - which it contains two newlines as `\n\n` in the string. I think the easiest way to deal with this is to split the filter into "one or more party members have yet to complete this duty" and parse the second sentence separately to narrow things down.

It's also possible they really are Progress-channel messages and not System-channel messages... so maybe I am filtering in the wrong channel after all - can you install the plugin `ChatTyper` while testing? It adds which channel the chat message comes from to messages which helps with debugging this. TBH I should probably add that to Tidy Chat's debug mode.

I've pushed a change to check the sentences separately - can you pull the changes and use the new build to test with?

**Comment by Dedoooo** [comment]
ah yeah i'm aware of the toast message i just chose to screenshot that instead of the chatlog since it was neater lol! i'll screenshot the chatlog next time.

> can you install the plugin `ChatTyper` while testing?
will do that now

**Comment by NadyaNayme** [comment]
I've added ChatTyper's primary functionality to Tidy Chat's debug mode. Was a whopping 1 line of code using information I already was passing through but not logging lol.

ChatTyper has the advantage of not spamming the chat with [TidyChat] [Debug] on almost every line - but Tidy Chat's debug mode now also includes which channel the message was sent to. Now I don't have to tell people to install ChatTyper - they can just flip Debug Mode on. :)

![image](https://user-images.githubusercontent.com/4567166/169222921-c89e99b8-bb90-4105-b7f0-44dc7ea4caca.png)

**Comment by Dedoooo** [comment]
Party Promotion ❌ 
![image](https://user-images.githubusercontent.com/93833740/169220383-579922e6-b818-4d09-8a9a-a5ba4763e7f1.png)
___________________________________
Local Linkshell Leader Promotion ✔️ 

Local Linkshell Leader Demote ❌ 
![image](https://user-images.githubusercontent.com/93833740/169223569-9bd4c714-efdf-4924-9121-733f30555788.png)

Local Linkshell Master ❌ 
![image](https://user-images.githubusercontent.com/93833740/169223749-430c2ced-c48d-4ed9-a46a-b92b7296e42f.png)

Else Local Linkshell promotion 
![image](https://user-images.githubusercontent.com/93833740/169223858-d8b287b5-73f2-45a1-85e2-d2ece25203e8.png)
___________________________________


Cross-World Linkshell Leader Promotion ✔️ 

___________________________________


First time player bonus announcement ✔️

**Comment by NadyaNayme** [comment]
So the Linkshell demotion is weird. " you are no longer a leader" is an exact text match with the filter and 313 is a System-channel message. So it should have worked.
![image](https://user-images.githubusercontent.com/4567166/169225110-8ebab2ff-fe57-43e4-a0b2-1e12abb5e566.png)

"Master privileges" was not in the original added list so it makes sense why it didn't m atch and neither was including _other_ people being promoted to leader (it was matching with including "you"). 

Party Promotion didn't match the regex but I tossed some extra `or` filters real fast. Probably need to do that for JP/DE/FR but maybe not. I don't care to do it right this second and will probably just wait til someone complains.

![image](https://user-images.githubusercontent.com/4567166/169225595-abe81dbd-4f23-4601-bfed-6e0068186d7e.png)

**Comment by Dedoooo** [comment]
i havent tried the" 2/1/0 chest" but hopefully it works! :) thanks a lot!

**Comment by Dedoooo** [comment]
i've got a question unrelated to all of this.
when using chat2's data base "display previous session" messages how does tidychat handle that? i've noticed the filter count has been getting huge really fast; does it filter everything over and over again every single time it populates the chat? @NadyaNayme 
right now i'm at 2million messages blocked if thats really big idk lol

**Comment by NadyaNayme** [comment]
Chat2's DB shouldn't be writing the messages to chat but just loading them - so nothing gets blocked by Tidy Chat. The blocked message count was bugged a few releases back where after 100 blocked messages in a zone every message would block 100+n because the current count wasn't being reset when adding to the total count.

When I changed the variable name that should have changed. Most people were in the 1,000,000+ blocked ranged due to this bug when in reality 1,000,000 was around 1,000~  messages.

I'm the dev and I'm only at 5,700~ messages blocked myself. :P
